src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingDiagnosticValidationPass.cs

```diff
@@ -65,6 +65,23 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
         // at all possible). Also worth noting the order has to be maintained in that case.
         if (!originalDiagnostics.SequenceEqual(changedDiagnostics, LocationIgnoringDiagnosticComparer.Instance))
         {
+            // Yes, these log messages look weird, but this is how structured logging works. The first parameter, the "template" is not
+            // supposed to change, or it causes lots of allocations. The second parameter is the "argument" which is supposed to change.
+            // In our case, we never use structured logging such that the argument name is important, so we just use the same one, and
+            // save some memory, and still log the expected values.
```

Comment:
I'm a little confused by this. This logging is `MS.Extensibility.Logging` right? Does it just mean the structured logging implementation requires this?

Reply:
Yeah. With structured logging if you just log a plain string, it's a memory/perf issue. It's not like string format.

This is the warning you get if you don't do this:
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2254